### PR TITLE
Overview of Microsoft SSIS

### LINKED ISSUE #5939: [Artificial Intelligence] Overview of Microsoft SSIS
#### Introduction
SQL Server Integration Service is a data storage solution that allows you to extract, manipulate, and load data quickly and easily. It makes it simple to transfer data from one database to another, such as SQL Server, Oracle, or an Excel spreadsheet. In this article, we will learn how SSIS works and the requirements of its integration service. In addition for better understanding, we'll have an example of both a data flow task and an execute SQL task

#### Key takeaways 
1. SSIS definition and how it works
2. SSIS tasks and packages
3. Advantages and disadvantages of SSIS
4. SSIS Best Practices

#### Article quality
The article will be unique as it will cover well-explained examples of how SSIS tasks and packages work. To add on, there will be SSIS best practices as an additional topic. 

#### References
N/A

### PR DESCRIPTION
**Pull request by Yakobo5**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5939

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Kimosop-Wakanda** [comment]
Greetings @ahmadmardeni1 , @hectorkambow .
I would like to review this article.

File: content/articles/microsoft-ssis/index.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
+### Overview of Microsoft SSIS
+
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:2]: Before the table of contents, write one or two introduction paragraphs showing what your article intends to cover. Look at other published articles like [this one](https://www.section.io/engineering-education/basics-of-means-end-analysis-in-ai/) to get an overview of how this is done.

File: content/articles/microsoft-ssis/index.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+### Introduction
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:3]: Rename this section 'Introduction to Microsoft SSIS and place it after the table of contents.

File: content/articles/microsoft-ssis/index.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+Microsoft SSIS (SQL Server Integration Services) is a tool developed by the Microsoft Company. It integrates, transforms, and migrates data in   Microsoft's SQL Server database. It may be applied in a wide range of integration activities, such as data purification and analysis and extraction, transformation, and loading (ETL) operations to upgrade data centers.
+
+Data Transformation Services, also known as DTS, is an older data management tool offered with SQL Server. Due to the advancing technology, it has been replaced by SSIS. The software included the Microsoft SQL Server database, and it used to change data from conventional databases and formats. XML, flat files, and Microsoft Excel are among the file formats recognized by SSIS. Many businesses use the program as a data management tool.
+
+### Table of Contents
+- [Introduction](#introduction)
+- [Key Takeaways](#key-takeaways)
+- [SSIS Components](#ssis-components)
+- [SSIS Tasks](#ssis-components)
+- [SSIS Packages](#ssis-packages)
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:13]: adjust this table of contents to correspond to the changes made in the document.

File: content/articles/microsoft-ssis/index.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+### Key Takeaways
+After reading this article, the reader should be able to understand the following:
+* SSIS components, tasks, and packages
+* How SSIS works
+* Advantages and disadvantages of using SSIS
+* SSIS best practices
+
+### SSIS Components
+1. Control Flow
+The control flow is the main component of the SSIS package. It acts as the SSIS's brain.  It assists in arranging the execution sequence for all of its components. Containers and tasks are controlled using priority restrictions inside the components.
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:30]:
```suggestion
- **Control Flow:** The control flow is the main component of the SSIS package. It acts as the SSIS's brain.  It assists in arranging the execution sequence for all of its components. Containers and tasks are controlled using priority restrictions inside the components.

File: content/articles/microsoft-ssis/index.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+2. Precedence Constraints
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:31]: use the example in no. 1 above and apply it throughout this section.

File: content/articles/microsoft-ssis/index.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+A precedence constraint is an SSIS component that instructs tasks to run in a specific sequence. Thus it describes the SSIS package's overall process. It directs the implementation of the two related activities by performing the target tasks according to the outcome of the previous task. 
+3. Tasks
+A 'Task' is defined as a discrete piece of activity in an SSIS package. It's similar to a programming language's method or function. Nonetheless, coding approaches are not used in SSIS. Rather, you'll create and customize surfaces using the drag-and-drop method.
+4. Containers
+The container is a component that organizes tasks into workgroups. It additionally enables you to specify variables and events controllers that should be in the range of that particular container, in addition to providing visual continuity.
+In SSIS, there are four different kinds of containers:
+  * A container for sequences.  A Sequence Container enables a user to gather and arrange secondary tasks perform operations, and allocate logs to the container.
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:38]: put the key words in bold.

File: content/articles/microsoft-ssis/index.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+  * Container for the Foreach Loop.  A Loop Container also permits looping. However, rather than utilizing a conditional expression, the loop is performed across a group of objects, such as files inside a folder.
+5. Data Flow
+The SSIS tool is mainly used to extort data from the server's storage, alter, and write it to a different location. While Control Flow is the main component of SSIS, Data Flow is its heart.
+6. SSIS Packages
+The 'package' is another vital aspect of SSIS. It comprises a set of tasks that run in a logical order. Primary restrictions are used to control the task's execution sequence. A package can assist a user in saving data to package catalog database on SQL Server. It can store as a .dtsx file, an organized format comparable to Reporting Services'.rdl files.
+7. Parameters
+With minor differences, parameters act similarly to variables. It's simple to take it out of the packaging. It can be specified as a set of values that are supplied before the package can begin.
+
+## SSIS Tasks
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:48]: change it to level three header. Look at other published articles to see how headers are formatted.

File: content/articles/microsoft-ssis/index.md [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+The following are the significant tasks included in Integration Services.
+
+**Data Flow Task**
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:51]: Use the suggestions in the section 'SSIS components' to format this section.

File: content/articles/microsoft-ssis/index.md [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+
+**Data sources and destinations.** It can also be referred to as data flow.  Sources and destinations are involved in extracting and loading data, alterations that alter and expand data. It also contains pathways that connect sources, conversions, and destinations make up a data flow. A Data Flow task should be included inside the package control flow before adding a data flow. A Data Flow task is an SSIS package component that produces, organizes, and executes data flows. Every Data Flow job in a package gets its replica of the data flow processor.
+
+**Connection managers.** Usually, a minimum of one connection manager is included inside a package. A connection manager connects between a package and a data source. It specifies the access string for obtaining the data used by the package's processes, transforms, and event controllers. 
+
+#### Package Functionality
+ Below are examples of additional objects which can be included in packages to give them advanced capabilities or enhance current functionality.
+
+**Event Handlers**
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:96]: For all the sub-sections in this section, use level 5 headers.

File: content/articles/microsoft-ssis/index.md [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+There are two types of variables supported by the Integration Services; system variables and user-defined variables. User-defined variables permit bespoke situations in packages, whereas system variables give essential information regarding package objects at program execution. 
+
+**Parameters**
+At the stage of package deployment, Integration Services parameters enable a user to set values to attributes inside packages. At the project level, users may design project parameters. They are utilized in routing any external input received by the project to any one of the project's packages. Package parameters allow users to change how a package runs without altering and re-deploying it.
+
+## How Microsoft SSIS Works
+
+SSIS is a tool used for two functions: data integration and workflow. The SSIS package is used to do various data transformations and workflow generation. The SSIS package is made up of three parts.
+
+**1. Operational data**.
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:115]: Use the suggested format in the section 'SSIS components'.

File: content/articles/microsoft-ssis/index.md [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+
+**Step one:** It all begins with a functional data warehouse, a database built to combine data from many sources to perform further actions on it.
+
+**Step two:** The extraction, transformation, and loading procedure are the next to be followed.
+
+**Step three:** The data warehouse collects information from various sources and organizes it for easy accessibility and usage.
+
+**Step four:** The data warehouse stores data to combine and handle multiple sources to solve the issues. As a result, it aids decision-making.
+
+
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:135]: remove the extra spaces.

File: content/articles/microsoft-ssis/index.md [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+
+
+### Advantages of SSIS 
+1. SSIS can combine data from multiple sources into a single package. We state that data sources can include customized or programmed interfaces.
+2. SSIS uses tricky data to access, such as FTP, HTTP, and Analysis services, among others.
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:140]: use consistent spacing.

File: content/articles/microsoft-ssis/index.md [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+### Disadvantages of SSIS
+1. Instead of getting posted to reporting services, Management Studio must view the package deployment result.
+
+2. If a user has many packages that require to be executed simultaneously, it will cause some trouble. It is because  SSIS uses a lot of RAM and interferes with SQL.
+
+3. When it comes to CPU allocations, it's also an issue when a user has many packages to execute simultaneously. The processor allocations among SQL and SSIS must ensure it is conducted correctly; SQL will have the dominant position, and SSIS will operate relatively slowly.
+
+#### SSIS Best Practices
+
+**1. Sorting**
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:162]: use the suggested format in the section 'SSIS components'.

File: content/articles/microsoft-ssis/index.md [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+
+**3. Limiting package names**
+
+If an SSIS package has a package name that is longer than one hundred characters is uploaded to the SQL Server, the name is trimmed down to one hundred characters, resulting in an execution error. As a result, package names should not exceed one hundred characters.
+
+**4. Configuration filter**
+
+It is an excellent practice. It utilizes the package name as the configuration filter for all package-specific configuration elements. It's helpful when there are a lot of packages with different setup options. 
+### Conclusion
+During the load operation, SSIS provides activities for transforming and validating data and transformations for inserting data into its destination. It's better to understand the various SSIS tasks and how they are utilized instead of writing a stored method in T-SQL to verify or alter data.
```
Comment (mbaabuones) [inline content/articles/microsoft-ssis/index.md:178]: The conclusion can be optimized by providing a summary of what was covered in the article. Your conclusion should have at least two paragraphs.

**Review by mbaabuones — CHANGES_REQUESTED** [review]
@Yakobo5 Thank you for submitting your first article. I take this opportunity to welcome you to the Section Engineering program. I have noticed that you have already uploaded your author bio and photo. That is commendable. The article is informative and well structured. I have suggested some minor changes that can improve the quality of this article. Please work on ALL the suggested changes carefully to avoid closure of this PR. When you finish revising the article, proofread it to ensure there are no grammatical or typographical errors. Ping me when you are done so that I can continue with the review process.

**Comment by Yakobo5** [comment]
Hello @mbaabuones, I have made the requested changes

**Comment by louisefindlay23** [comment]
Plagiarism check complete

**Comment by hectorkambow** [comment]
@mbaabuones  

Any update on this PR? :rocket

**Review by mbaabuones — APPROVED** [review]

**Comment by mbaabuones** [comment]
> @mbaabuones
> 
> Any update on this PR? 🚀

@hectorkambow This article is in the tech review.

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**